### PR TITLE
 Add skeleton for awsxrayproxy extension.

### PR DESCRIPTION
**Pull request by anuraaga**
**Description:**

Currently, there is a proxy server for the AWS X-Ray API in the awsxrayreceiver component. This is to allow use of the AWS X-Ray SDK's remote sampling feature, while only requiring actual AWS authentication set up on the collector, not the user's app.

We've implemented the first support for X-Ray remote sampling in OpenTelemetry.

https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/aws-xray

So now we need a way to do the same for users of OTel. It's actually already possible if enables the xrayreceiver, but awkward because it's only used for this proxy and not needed for export. Instead, OTel users use the otlp receiver with the awsxrayexporter.

Because the proxy has no correlation to the export pipeline, we'd like to move it to an extension. Then it can be easily enabled in conjunction with the awsxrayexporter. Or any other exporter really, it would even be possible to use X-Ray sampling along with e.g. zipkin exporter if a user wanted to do so.

This PR copies over the skeleton of the proxy code from the awsxrayreceiver with no significant change. The only one change is removing `local_mode` from the Config. I've found it mostly adds confusion without any real value.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Unit tests

**Documentation:** README

### CODE REVIEW

**Comment by anuraaga** [comment]
@mxiamxia @Aneurysm9 @willarmiros @bhautikpip If you can help review would be great, thanks!

File: extension/awsxrayproxy/config.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+// Config defines the configuration for an AWS X-Ray proxy.
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// endpoint is the TCP address and port on which this receiver listens for
```
Comment (bhautikpip) [inline extension/awsxrayproxy/config.go:27]: Can we probably edit the description? I think in this case we will be using X-Ray service APIs like `GetSamplingRules` and `GetSamplingTargets` via this proxy right? So we probably don't need X-Ray SDK in this case.

**Review by bhautikpip — COMMENTED** [review]
Just some minor comments otherwise looks good so approving. Thanks for making this change!

**Review by bhautikpip — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: extension/awsxrayproxy/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+The default configurations below are based on the [default configurations](https://github.com/aws/aws-xray-daemon/blob/master/pkg/cfg/cfg.go#L99) of the existing X-Ray Daemon.
+
+### endpoint (Optional)
+The TCP address and port on which this proxy listens for requests.
+
+Default: `0.0.0.0:2000`
+
+### proxy_address (Optional)
+Defines the proxy address that this proxy forwards HTTP requests to AWS X-Ray backend through. If left unconfigured, requests will be sent directly.
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply extension/awsxrayproxy/README.md]: Thanks! By the way, do you think this flag has no use case now that X-Ray supports VPC endpoints? Just wondering.

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@anuraaga please also add CODEOWNERs for this component.

**Pull request merged**